docs/design/flow-monitoring/flow-latency-fsm.puml

```diff
@@ -0,0 +1,45 @@
+@startuml
+title FlowLatency FSM
+
+legend top left
+    process tick:
+        if last event is timed out
+            send corresponding stable-* event
+        save last event type (healthy | tier1failed | tier2failed) and time
+    unstable enter:
+        if event latency exceeds the threshold
+            send corresponding stable-* event
+        save last event type (healthy | tier1failed | tier2failed) and time
+endlegend
+
+[*] --> Healthy : flow latency within SLA
+[*] --> Tier1Failed: flow latency SLA tier 1 is violated
+[*] --> Tier2Failed: flow latency SLA tier 2 is violated
+
+Healthy: enter / save current flow latency to DB
+Healthy: tick / save current flow latency to DB
+Healthy: healthy / update current latency
+
+Healthy --> Unstable : tier1failed / tier2failed
+
+Tier1Failed: enter / save current flow latency to DB
+Tier1Failed: tick / save current flow latency to DB
+Tier1Failed: tier1failed / update current latency
+
+Tier1Failed --> Unstable : healthy / tier2failed
+
+Tier2Failed: enter / save current flow latency to DB
+Tier2Failed: tick / save current flow latency to DB
+Tier2Failed: down / update current latency
+
+Tier2Failed --> Unstable : healthy / tier1failed
+
+Unstable: enter / unstable enter
+Unstable: tick / process tick
+Unstable: stable-* / send reroute or sync request if needed
```

Comment:
Add handling of healthy/tier1fail/tail2fail events here, as far a I understood "comment" this handling really happens, but diagram do not show this and it is confusing.

Reply:
Added.